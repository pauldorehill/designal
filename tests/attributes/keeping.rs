use designal::Designal;

#[derive(Designal)]
struct HumanBean {
    #[forbid(dead_code)]
    taste: String,
}

#[derive(Designal)]
#[forbid(dead_code)]
struct HumanBean1 {
    crunch: String,
}

fn main() {}
