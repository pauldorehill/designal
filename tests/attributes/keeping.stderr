error: field is never read: `taste`
 --> $DIR/keeping.rs:6:5
  |
6 |     taste: String,
  |     ^^^^^^^^^^^^^
  |
note: the lint level is defined here
 --> $DIR/keeping.rs:5:14
  |
5 |     #[forbid(dead_code)]
  |              ^^^^^^^^^

error: field is never read: `taste`
 --> $DIR/keeping.rs:6:5
  |
6 |     taste: String,
  |     ^^^^^^^^^^^^^

error: struct is never constructed: `HumanBean1`
  --> $DIR/keeping.rs:11:8
   |
11 | struct HumanBean1 {
   |        ^^^^^^^^^^
   |
note: the lint level is defined here
  --> $DIR/keeping.rs:10:10
   |
10 | #[forbid(dead_code)]
   |          ^^^^^^^^^
